src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -280,11 +280,38 @@ private Action<TRow> GenerateSetter(IRow input, int index, InternalSchemaDefinit
                     if (fieldType.GetElementType() == typeof(string))
                     {
                         Ch.Assert(colType.ItemType.IsText);
-                        return CreateVBufferToStringArraySetter(input, index, poke, peek);
+                        return CreateVBufferSetter<DvText, string>(input, index, poke, peek, (x) => x.ToString());
+                    }
+                    else if (fieldType.GetElementType() == typeof(bool))
+                    {
+                        Ch.Assert(colType.ItemType.IsBool);
+                        return CreateVBufferSetter<DvBool, bool>(input, index, poke, peek, (x) => Convert.ToBoolean(x.RawValue));
+                    }
+                    else if (fieldType.GetElementType() == typeof(int))
+                    {
+                        Ch.Assert(colType.ItemType == NumberType.I4);
+                        return CreateVBufferSetter<DvInt4, int>(input, index, poke, peek, (x) => x.RawValue);
+                    }
+                    else if (fieldType.GetElementType() == typeof(short))
```

Comment:
If we're doing this with `int`, the explicit casts would throw in the case of missing values, whereas things like this will just silently pass `int.MinValue`.

Reply:
That is, I think we *do* want to throw here, if someone makes the mistake of using a type that doesn't support missing values, over data that actually has missing values.